src/widget/scroll_view.rs

```diff
@@ -47,41 +44,26 @@ impl ScrollView {
     }
 }
 
+// TODO: scroll bars
 impl Widget for ScrollView {
-    fn event(&mut self, cx: &mut EventCx, event: &RawEvent) {
-        // TODO: scroll wheel + click-drag on scroll bars
-        let offset = Vec2::new(0.0, self.offset);
-        let child_event = match event {
-            RawEvent::MouseDown(mouse_event) => {
-                let mut mouse_event = mouse_event.clone();
-                mouse_event.pos += offset;
-                RawEvent::MouseDown(mouse_event)
-            }
-            RawEvent::MouseUp(mouse_event) => {
-                let mut mouse_event = mouse_event.clone();
-                mouse_event.pos += offset;
-                RawEvent::MouseUp(mouse_event)
-            }
-            RawEvent::MouseMove(mouse_event) => {
-                let mut mouse_event = mouse_event.clone();
-                mouse_event.pos += offset;
-                RawEvent::MouseMove(mouse_event)
-            }
-            RawEvent::MouseWheel(mouse_event) => {
-                let mut mouse_event = mouse_event.clone();
-                mouse_event.pos += offset;
-                RawEvent::MouseWheel(mouse_event)
-            }
-            _ => event.clone(),
-        };
-        self.child.event(cx, &child_event);
+    fn event(&mut self, cx: &mut EventCx, event: &Event) {
+        // Pass event through to child
+        self.child.event(cx, event);
+
+        // Handle scroll wheel events
         if !cx.is_handled() {
-            if let RawEvent::MouseWheel(mouse) = event {
-                let new_offset = (self.offset + mouse.wheel_delta.y).max(0.0);
+            if let Event::MouseWheel(mouse) = event {
```

Comment:
Out of a UX perspective it would be interesting (not entirely sure how this is done in web, but I think it's similar as here?), how to handle nested scroll-views.

When I'm getting this right, the nested scroll view handles the scroll event, unless it doesn't scroll (e.g. is at the end).

Just thinking out loud, not suggesting anything.

Reply:
I believe that ideally it would be a little more sophisticated than that, and that the inner view could handle part of a scroll event (some of the pixel delta) and leave the rest for an outer scroll view (perhaps by making the event mutable?). But you suggestion would probably cover the basics.